### PR TITLE
Incognito client uses a single highway as a message forwarder

### PR DESCRIPTION
**Pull request by 0xbunyip**
Changes:

1. peerv2/conn_manager: manages connection to highway => all messages are sent to highway
2. masternode: broadcasts new committee to highway whenever there's a substitution
3. peerstatev2: receives (same shard) peerstate from highway to know which blocks to request from other peers (note: cross-shard peerstate is removed)
4. block_requester: grpc client, used to request blocks from other peers
5. block_provider: grpc server, serves requests for blocks from highway
6. server.go: PushMessageGetBlock... => use block_requester to request blocks instead of broadcasting Get messages

### CODE REVIEW

File: blockchain/syncblockutils.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func getBlkNeedToGetByHash(prefix string, blksHash []common.Hash, cachedItems ma
 	return blocksNeedToGet
 }
 
-func getBlkNeedToGetByHeight(prefix string, fromHeight uint64, toHeight uint64, cachedItems map[string]cache.Item, poolItems []uint64, peerID libp2p.ID) map[uint64]uint64 {
+func getBlkNeedToGetByHeight(prefix string, fromHeight uint64, toHeight uint64, cachedItems map[string]cache.Item, poolItems []uint64) map[uint64]uint64 {
```
Comment (thaibao-incognito) [inline blockchain/syncblockutils.go]: This function is the same content with getBlkNeedToGetBySpecificHeightv2

Comment (0xbunyip) [reply blockchain/syncblockutils.go]: @0xakk0r0kamui

Comment (0xakk0r0kamui) [reply blockchain/syncblockutils.go]: Ok, my bad.

File: blockchain/beaconprocess.go [hunk 2]
```diff
@@ -247,6 +256,16 @@ func (blockchain *BlockChain) InsertBeaconBlock(beaconBlock *BeaconBlock, isVali
...
 	go blockchain.config.PubSubManager.PublishMessage(pubsub.NewMessage(pubsub.NewBeaconBlockTopic, beaconBlock))
 	go blockchain.config.PubSubManager.PublishMessage(pubsub.NewMessage(pubsub.BeaconBeststateTopic, blockchain.BestState.Beacon))
+
+	// For masternode: broadcast new committee to highways
+	if notifyHighway {
+		go blockchain.config.Highway.BroadcastCommittee(
+			blockchain.config.ChainParams.Epoch,
+			newBeaconCommittee,
+			newAllShardCommittee,
+			newAllShardPending,
```
Comment (0xakk0r0kamui) [inline blockchain/beaconprocess.go]: TEst

File: incognitokey/highway_committee.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package incognitokey
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply incognitokey/highway_committee.go]: done

File: log.go [hunk 4]
```diff
@@ -37,8 +37,8 @@ var (
 	rpcLogger              = backendLog.Logger("RPC Log", false)
 	rpcServiceLogger       = backendLog.Logger("RPC service Log", false)
 	rpcServiceBridgeLogger = backendLog.Logger("RPC service DeBridge Log", false)
-	netsyncLogger          = backendLog.Logger("Netsync Log", true)
-	peerLogger             = backendLog.Logger("Peer Log", true)
+	netsyncLogger          = backendLog.Logger("Netsync Log", false)
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply log.go]: done

File: peerv2/block_provider.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
+package peerv2
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply peerv2/block_provider.go]: done

File: peerv2/block_requester.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
+package peerv2
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply peerv2/block_requester.go]: done

File: peerv2/conn_manager.go [hunk 7]
```diff
@@ -0,0 +1,543 @@
+package peerv2
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply peerv2/conn_manager.go]: done

File: netsync/netsync.go [hunk 8]
```diff
@@ -409,7 +409,7 @@ func (netSync *NetSync) handleMessageShardBlock(msg *wire.MessageBlockShard) {
 }
 
 func (netSync *NetSync) handleMessageCrossShard(msg *wire.MessageCrossShard) {
-	Logger.log.Debug("Handling new message CrossShard")
+	Logger.log.Error("Handling new message CrossShard")
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply netsync/netsync.go]: done

File: server.go [hunk 9]
```diff
@@ -632,10 +684,11 @@ func (serverObj Server) Start() {
 		return
 	}
 	Logger.log.Debug("Starting server")
+	/* --- Checkforce update code --- TODO uncomment this
 	if serverObj.chainParams.CheckForce {
 		serverObj.CheckForceUpdateSourceCode()
-	}
-	if cfg.IsTestnet() {
+	}*/
+	if cfg.TestNet == "testnet" {
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply server.go]: fixed

File: server.go [hunk 10]
```diff
@@ -632,10 +684,11 @@ func (serverObj Server) Start() {
 		return
 	}
 	Logger.log.Debug("Starting server")
+	/* --- Checkforce update code --- TODO uncomment this
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply server.go]: fixed

File: blockchain/messages.go [hunk 11]
```diff
@@ -182,17 +251,19 @@ func (blockchain *BlockChain) OnShardToBeaconBlockReceived(block *ShardToBeaconB
 			}
 		}
 		if from != 0 && to != 0 {
-			fmt.Printf("Message/SyncBlkShardToBeacon, from %+v to %+v \n", from, to)
+			Logger.log.Infof("[hy] Message/SyncBlkShardToBeacon, from %+v to %+v \n", from, to)
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply blockchain/messages.go]: done

File: consensus/blsbft/bftactor.go [hunk 12]
```diff
@@ -345,10 +345,10 @@ func (e *BLSBFT) enterVotePhase() {
 
 func (e *BLSBFT) enterNewRound() {
 	//if chain is not ready,  return
-	if !e.Chain.IsReady() {
-		e.RoundData.State = ""
-		return
-	}
+	// if !e.Chain.IsReady() {
```
[the remark this answers is not in the document]
Comment (0xbunyip) [reply consensus/blsbft/bftactor.go]: fixed

File: blockchain/syncblockutils.go (hunk 1 above)
Comment (0xbunyip) [reply blockchain/syncblockutils.go]: removed v2, keep prefix

**Pull request merged**